src/hooks/useStakingValidator.ts

```diff
@@ -42,9 +49,54 @@ export const useStakingValidator = () => {
 
     const response = await getValidators();
 
-    const filteredValidators = response?.validators.filter((validator) =>
+    // Filter out jailed and unbonded validators
+    const availableValidators =
+      response?.validators.filter(
+        (validator: Validator) => validator.status === 3 && validator.jailed === false
+      ) ?? [];
+
+    // Sort validators 1/ in ascending commission and 2/ by descending stake weight
+    const sortByCommission = (validatorA: Validator, validatorB: Validator): number => {
+      return MustBigNumber(validatorA.commission?.commissionRates?.rate ?? 0).gt(
+        MustBigNumber(validatorB.commission?.commissionRates?.rate ?? 0)
+      )
+        ? 1
+        : -1;
+    };
+
+    const sortByCommissionAndStakeWeight = (
+      validatorA: Validator,
+      validatorB: Validator
+    ): number => {
+      if (
+        (validatorA.commission?.commissionRates?.rate ?? 0) ===
+        (validatorB.commission?.commissionRates?.rate ?? 0)
+      ) {
+        return MustBigNumber(validatorA.delegatorShares).gt(
+          MustBigNumber(validatorB.delegatorShares)
+        )
+          ? -1
+          : 1;
+      }
+      return 0;
+    };
+
+    availableValidators.sort(sortByCommission);
+    availableValidators.sort(sortByCommissionAndStakeWeight);
```

Comment:
i believe we can achieve this by sorting by stakeweight desc and then sorting by commission asc after, if that feels easier to read for you

Reply:
we can potentially combine the sorting into a single sort function too I think, but maybe thats a bit more confusing.